src/metabase/query_processor/middleware/parameters/mbql.clj

```diff
@@ -28,17 +38,14 @@
         (not (string? param-value)))
     param-value
 
-    ;; if PARAM-VALUE contains a period then convert to a Double
-    (re-find #"\." param-value)
-    (Double/parseDouble param-value)
-
-    ;; otherwise convert to a Long
     :else
-    (Long/parseLong param-value)))
+    (to-numeric param-value)))
```

Comment:
I ended up not parsing anything here. I suspect we need to figure out how to do field-filters from query strings and ensure it works